File: Python/project/src/Lists/KeepMultiFoundValuesByTwo/KeepMultiFoundValuesByTwo.py
from typing import List

def main():
    nums = [5, 3, 6, 1, 12]
    original = 3

    solution = Solution()
    output = solution.findFinalValue(nums, original)
    print("Find Final Value: " + str(output))


class Solution:
    def findFinalValue(self, nums: List[int], original: int) -> int:
        print("Finding Final Value")

        val = original
        nums.sort()
        print("Sorted: " + str(nums))

        while True:

            if val in nums:
                i = nums.index(val)
            else:
                break

            current = nums[i]
            print("i: " + str(i) + "    current: " + str(current))

            val = current * 2

        return val

if __name__ == "__main__":
    main()

File: Java/src/main/java/Lists/KeepMultiFoundValuesByTwo/KeepMultiFoundValuesByTwo.java
package Lists.KeepMultiFoundValuesByTwo;

import java.util.Arrays;

public class KeepMultiFoundValuesByTwo {

    public static void main(String [] args) {
        int [] nums = {5,3,6,1,12};
        int original = 3;
        System.out.println("find Find Value: " + findFinalValue(nums, original));

        int [] nums1 = {2,7,9};
        int original1 = 4;
        System.out.println("find Find Value: " + findFinalValue(nums1, original1));

    }

    public static int findFinalValue(int[] nums, int original) {
        int val = original;
        Arrays.sort(nums);

        while (true) {
            int i = Arrays.binarySearch(nums, val);
            if (i < 0)
                break;

            int current = nums[i];
            System.out.println("i: " + i + "    current: " + current);


            val = current * 2;
        }

        return val;
    }
}
